streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractStream.java

```diff
@@ -31,6 +31,8 @@
     protected final Set<String> sourceNodes;
 
     public AbstractStream(KStreamBuilder topology, String name, Set<String> sourceNodes) {
+        assert sourceNodes != null && !sourceNodes.isEmpty() : "parameter <sourceNodes> must not be null or empty";
```

Comment:
@guozhangwang replaces assert with exception. Not sure about extra unit test -- we do have different unit and integration tests, with different operators. Do you think, this implicit testing is not good enough? Adding some "random" operator combination is no exhaustive test either... I also stepped through the code and reviews all constructor calls (found one issue in `merge()` and added additional check there. For future changes, we might want to add more test gradually.